src/hooks/plan_tree_walker.c

```diff
@@ -0,0 +1,106 @@
+#include <postgres.h>
+
+#include <miscadmin.h>
+#include <nodes/nodeFuncs.h>
+#include <nodes/parsenodes.h>
+#include <nodes/plannodes.h>
+
+bool plan_tree_walker_util(Plan *plan, bool (*walker_func)(Plan *plan, void *context), void *context)
+{
+    if(walker_func(plan->targetlist, context)) return true;
+    if(walker_func(plan->qual, context)) return true;
+    if(walker_func(plan->lefttree, context)) return true;
+    if(walker_func(plan->righttree, context)) return true;
+    if(walker_func(plan->initPlan, context)) return true;
+    return false;
+}
+
+bool plan_tree_walker(Plan *plan, bool (*walker_func)(Plan *plan, void *context), void *context)
+{
+    check_stack_depth();
```

Comment:
what does this do?

Reply:
This is in Postgres's expression_tree_walker function. It checks that we aren't going into too deep a recursion. So would be triggered for an unreasonable complex query, or if there's somehow an infinite loop.